diff '--color=auto' -Naur a/bin/amavisd b/bin/amavisd
--- a/bin/amavisd	2026-01-09 18:17:30.326992298 +0100
+++ b/bin/amavisd	2026-01-09 18:17:43.464179021 +0100
@@ -5,6 +5,8 @@
 ### #!/usr/bin/perl -d:NYTProf
 ###   NYTPROF=start=no:addpid=1:forkdepth=1 amavisd -m 5 foreground
 
+use lib $ENV{PERL5LIB} =~ /([^:]+)/g; # Untaint PERL5LIB so modules can be used
+
 #------------------------------------------------------------------------------
 # This is amavis.
 # It is an interface between a message transfer agent (MTA) and virus
diff '--color=auto' -Naur a/bin/amavisd-release b/bin/amavisd-release
--- a/bin/amavisd-release	2026-01-09 18:17:30.327048541 +0100
+++ b/bin/amavisd-release	2026-01-09 18:17:43.464391905 +0100
@@ -88,7 +88,7 @@
   $log_level = 1;
 # $socketname = '127.0.0.1:9998';
 # $socketname = '[::1]:9998';
-  $socketname = '/var/amavis/amavisd.sock';
+  $socketname = '/opt/zextras/data/amavisd/amavisd.sock';
 
 ### END OF USER CONFIGURABLE
 }
diff '--color=auto' -Naur a/bin/amavisd-status b/bin/amavisd-status
--- a/bin/amavisd-status	2026-01-09 18:17:30.327154720 +0100
+++ b/bin/amavisd-status	2026-01-09 18:17:43.464523520 +0100
@@ -48,6 +48,7 @@
 use warnings;
 use warnings FATAL => qw(utf8 void);
 no warnings 'uninitialized';
+use lib "/opt/zextras/common/lib/perl5";
 
 use Errno qw(ESRCH ENOENT);
 use POSIX qw(strftime);
diff '--color=auto' -Naur a/bin/amavis-mc b/bin/amavis-mc
--- a/bin/amavis-mc	2026-01-09 18:17:30.326942894 +0100
+++ b/bin/amavis-mc	2026-01-09 18:17:43.464658648 +0100
@@ -48,6 +48,7 @@
 use warnings;
 use warnings FATAL => qw(utf8 void);
 no warnings 'uninitialized';
+use lib "/opt/zextras/common/lib/perl5";
 
 use vars qw($VERSION);  $VERSION = 2.008002;
 
@@ -73,16 +74,16 @@
 
 ### USER CONFIGURABLE:
 
-$daemon_user   = 'vscan';
-@daemon_groups = 'vscan';
+$daemon_user   = 'zextras';
+@daemon_groups = 'zextras';
 
-$pid_file = '/var/amavis/amavis-mc.pid';
+$pid_file = '/run/carbonio/amavis-mc.pid';
 
 $log_level = 0;
 $syslog_ident = 'amavis-mc';
 $syslog_facility = LOG_MAIL;
 
-@path = qw(/usr/local/sbin /usr/local/bin /usr/sbin /sbin /usr/bin /bin);
+@path = qw(/opt/zextras/common/sbin /usr/local/sbin /usr/local/bin /usr/sbin /sbin /usr/bin /bin);
 
 @services = (
   { cmd => 'amavis-services msg-forwarder' },
diff '--color=auto' -Naur a/lib/Amavis/rfc2821_2822_Tools.pm b/lib/Amavis/rfc2821_2822_Tools.pm
--- a/lib/Amavis/rfc2821_2822_Tools.pm	2026-01-09 18:17:30.329647054 +0100
+++ b/lib/Amavis/rfc2821_2822_Tools.pm	2026-01-09 18:17:43.464845520 +0100
@@ -515,6 +515,9 @@
     my(@dkeys); my $d = $domain;
     for (;;) {            # (@).sub.example.com (@).example.com (@).com (@).
       push(@dkeys, $prepend_to_domain.'.'.$d);
+      if ($d ne '') {
+	push(@dkeys, $d);
+      }
       last  if $d eq '';
       $d = ($d =~ /^([^.]*)\.(.*)\z/s) ? $2 : '';
     }
diff '--color=auto' -Naur a/lib/Amavis.pm b/lib/Amavis.pm
--- a/lib/Amavis.pm	2026-01-09 18:17:30.327665229 +0100
+++ b/lib/Amavis.pm	2026-01-09 18:18:20.328097909 +0100
@@ -443,7 +443,7 @@
     P => sub {$MSGINFO->partition_tag}, # SQL partition tag
     partition_tag => sub {$MSGINFO->partition_tag},  # synonym for %P
     q => sub { my $q = $MSGINFO->quarantined_to;
-               $q && [map { my $m=$_; $m=~s{^\Q$QUARANTINEDIR\E/}{}; $m } @$q];
+               $q && [map { my($m)=$_; $m=~s{^\Q$QUARANTINEDIR\E/}{}; $m=~s/^march.*\.archive$//; $m } @$q];
              },  # list of quarantine mailboxes
     v => sub { !defined $av_output ? undef     # anti-virus scanner output
                  : [split(/[ \t]*\r?\n/, $av_output)]},
@@ -889,7 +889,7 @@
           sort map { my $s = $_; $s =~ s/\.pm\z//; $s =~ s{/}{::}g; $s }
                grep(/\.pm\z/, keys %INC)) {
     next  if !grep($_ eq $m, qw(Amavis::Conf
-      Archive::Tar Archive::Zip Compress::Zlib Compress::Raw::Zlib
+      Archive::Zip Compress::Zlib Compress::Raw::Zlib
       Convert::TNEF Convert::UUlib File::LibMagic
       MIME::Entity MIME::Parser MIME::Tools Mail::Header Mail::Internet
       Digest::MD5 Digest::SHA Digest::SHA1 Crypt::OpenSSL::RSA
@@ -5970,6 +5970,15 @@
     }
   }  # endfor per_recip_data
 
+  if (defined($qar_method) && $qar_method ne '') {  # archiving quarantine on sender
+    my($sender) = $msginfo->sender;
+    if ($sender ne '') {
+      my($q) = lookup(0,$sender,@{ca('archive_quarantine_to_maps')});
+      $q = $sender  if $q ne '' && $qar_method =~ /^bsmtp:/i;  # orig.recip
+      push(@q_tuples, [$qar_method,$q,'Arch'])  if defined $q && $q ne '';
+    }
+  }
+
   if ($ccat == CC_SPAM) {
     my $sqbsm = ca('spam_quarantine_bysender_to_maps');
     if (@$sqbsm) {  # by-sender spam quarantine (hardly useful, rarely used)
@@ -7010,7 +7019,7 @@
     push(@{/^no_conf_file_writable_check\z/ ? \@known : \@unknown}, $_)
       for grep($i_know_what_i_am_doing{$_}, keys %i_know_what_i_am_doing);
     $unknown[0] = 'unknown: ' . $unknown[0]  if @unknown;
-    warn sprintf("I know what I'm doing: %s\n", join(', ',@known,@unknown));
+    #warn sprintf("I know what I'm doing: %s\n", join(', ',@known,@unknown));
   }
 
   # deal with debugging early, based on a command line arg
