diff '--color=auto' -Naur a/Makefile.PL b/Makefile.PL
--- a/Makefile.PL	2026-04-24 09:17:25.289206656 +0200
+++ b/Makefile.PL	2026-04-24 09:20:20.044031558 +0200
@@ -4,10 +4,7 @@
 
 use 5.006;
 use ExtUtils::MakeMaker 6.64;
-use Crypt::OpenSSL::Guess qw(openssl_inc_paths openssl_lib_paths openssl_version);
 
-my ($major, $minor, $patch) = openssl_version(); 
-print "OpenSSL version: $major.$minor $patch", "\n";
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 
@@ -16,6 +13,10 @@
   $libs = $libs . ' -lz';
 }
 
+my $KERBEROS_INCLUDE = "/usr/kerberos/include";
+
+my $include = -d $KERBEROS_INCLUDE ? "-I${KERBEROS_INCLUDE}" : "";
+
 WriteMakefile(
     'NAME'             => 'Crypt::OpenSSL::RSA',
     'AUTHOR'           => 'Ian Robertson <iroberts@cpan.org>',
@@ -36,10 +37,8 @@
         'Test::More' => 0,
     },
     'OBJECT' => 'RSA.o',
-    'LIBS'   => [openssl_lib_paths() . $libs],
-    'LDDLFLAGS' => openssl_lib_paths() . ' ' . $Config{lddlflags},
-    'INC' => openssl_inc_paths(),    # e.g., '-I/usr/include/other'
-
+    'LIBS'    => ($^O eq 'MSWin32') ? ['-lssl32 -leay32'] : ['-lssl -lcrypto'],
+    'INC'	=> $include,
     'dist'  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
     'clean' => { FILES    => 'Crypt-OpenSSL-RSA-*' },
     'META_MERGE' => {
@@ -50,9 +49,6 @@
                 version => MM->parse_version('RSA.pm'),
             },
         },
-        configure_requires => {
-            'Crypt::OpenSSL::Guess' => '0.11',
-        },
         resources => {
             'license'    => ['https://dev.perl.org/licenses/'],
             'homepage'   => 'https://github.com/cpan-authors/Crypt-OpenSSL-RSA',
